### PR TITLE
Ramesh dev

### PR DESCRIPTION
**Pull request by RameshGKR**
Added function to return derivative

### CODE REVIEW

File: tests/derivative_test.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    /* derivative of a variable w.r.t itself */
+    test_derivative(x, "x", "1");
+
+    /* derivative of a variable w.r.t another variable */
+    test_derivative(x, "y", "0");
+
+    /* derivative of a simple expression: x^2 */
+    test_derivative(newExpOp(EXP_EXP_OP, x, newExpLeaf(EXP_NUM, "2")), "x", "((2 * 1) * (x^(2 - 1)))");
+
```
Comment (gaperez64) [inline tests/derivative_test.c:48]: I would add a new test for something like the derivative of x^3 + 42 * x * x + 10 * x - y. If that works, I'll be satisfied :)

**Comment by gaperez64** [comment]
The tests fail on macos, seemingly because `<string.h>` is missing in `funexp.c`?

File: tests/integral_test.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    
+    /* integral of x w.r.t itself */
+    test_integral(x, "x", "(0.5 * (x^2))");
+
+    /* integral of x w.r.t another variable */
+    test_integral(x, "y", "(x * y)");
+    
+    /* integral of a simple expression: x^2 */
+    test_integral(newExpOp(EXP_EXP_OP, x, newExpLeaf(EXP_NUM, "3")), "x", "((1 / (3 + 1)) * (x^(3 + 1)))");
+
```
Comment (gaperez64) [inline tests/integral_test.c:49]: Here too I would like to see an integral of a more complex function like the one I proposed for derivative

**Comment by RameshGKR** [comment]
> The tests fail on macos, seemingly because `<string.h>` is missing in `funexp.c`?

Indeed the issue seems to be missing `<string.h>`, should be fixed now

**Comment by gaperez64** [comment]
The tests still do not pass, now because of style problems.

**Comment by RameshGKR** [comment]
> Nope, style checks still not passing. Also, the two proposed tests above are not yet added.

I'll sort out the clang style issue first; pushed the style changes for the two tests

**Comment by gaperez64** [comment]
The style checks pass now.

**Comment by RameshGKR** [comment]
> The style checks pass now.

Nice. In future, before push, the code will be checked for correct style using, for example, `clang-format -i fun.c`.
Alternatively, we could set up a Git pre-commit hook that automatically formats the code before the commit is made.

File: tests/derivative_test.c (hunk 1 above)
Comment (RameshGKR) [reply tests/derivative_test.c:63]: Completed

File: tests/integral_test.c (hunk 2 above)
Comment (RameshGKR) [reply tests/integral_test.c]: Completed in pull request#2

**Pull request merged**